store/bigtable/bigtable.go

```diff
@@ -0,0 +1,470 @@
+	errCtxCanceled    = errors.New("context canceled")
+
+	// metric store.bigtable.get.exec is the duration of getting from bigtable store
+	btblGetExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.get.exec")
+	// metric store.bigtable.get.wait is the duration of the get spent in the queue
+	btblGetWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.get.wait")
+	// metric store.bigtable.put.exec is the duration of putting in bigtable store
+	btblPutExecDuration = stats.NewLatencyHistogram15s32("store.bigtable.put.exec")
+	// metric store.bigtable.put.wait is the duration of a put in the wait queue
+	btblPutWaitDuration = stats.NewLatencyHistogram12h32("store.bigtable.put.wait")
+	// metroc store.bigtable.get.error is the count of reads that failed
+	btblReadError = stats.NewCounter32("store.bigtable.get.error")
+
+	// metric store.bigtable.chunks_per_row is how many chunks are retrieved per row in get queries
+	btblChunksPerRow = stats.NewMeter32("store.bigtable.chunks_per_row", false)
+	// metric store.bigtable.rows_per_response is how many rows come per get response
+	btblRowsPerResponse = stats.NewMeter32("store.bigtable.rows_per_response", false)
+
+	// metric store.bigtable.chunk_operations.save_ok is counter of successful saves
+	chunkSaveOk = stats.NewCounter32("store.bigtable.chunk_operations.save_ok")
+	// metric store.bigtable.chunk_operations.save_fail is counter of failed saves
+	chunkSaveFail = stats.NewCounter32("store.bigtable.chunk_operations.save_fail")
+	// metric store.bigtable.chunk_size.at_save is the sizes of chunks seen when saving them
+	chunkSizeAtSave = stats.NewMeter32("store.bigtable.chunk_size.at_save", true)
+	// metric store.bigtable.chunk_size.at_load is the sizes of chunks seen when loading them
+	chunkSizeAtLoad = stats.NewMeter32("store.bigtable.chunk_size.at_load", true)
+	// metric store.bigtable.byte_size.at_save is the number of chunk bytes saved in each bulkApply
+	chunkBytesAtSave = stats.NewMeter32("store.bigtable.chunk_bytes.at_save", true)
+)
+
+func formatRowKey(key schema.AMKey, ts uint32) string {
+	return fmt.Sprintf("%s_%d", key.MKey.String(), ts/Month_sec)
+}
+
+func formatFamily(ttl uint32) string {
+	return fmt.Sprintf("ret_%d", ttlToHours(ttl))
+}
+
+func PrepareChunkData(span uint32, data []byte) []byte {
+	chunkSizeAtSave.Value(len(data))
+	version := chunk.FormatStandardGoTszWithSpan
+	buf := new(bytes.Buffer)
+	binary.Write(buf, binary.LittleEndian, version)
+
+	spanCode, ok := chunk.RevChunkSpans[span]
+	if !ok {
+		// it's probably better to panic than to persist the chunk with a wrong length
+		panic(fmt.Sprintf("Chunk span invalid: %d", span))
+	}
+	binary.Write(buf, binary.LittleEndian, spanCode)
+	buf.Write(data)
+	return buf.Bytes()
+}
+
+func mutationFromWriteRequest(cwr *mdata.ChunkWriteRequest) (*bigtable.Mutation, int) {
+	mut := bigtable.NewMutation()
+	family := formatFamily(cwr.TTL)
+	column := "raw"
+	parts := strings.SplitN(cwr.Key.String(), "_", 2)
```

Comment:
the better way to do this is just look at the archive. if `cwr.Key.Archive == 0` then it's raw, otherwise it's a rollup.
you can get the MKey string and archive string separately
`cwr.Key.MKey.String()` and ` cwr.Key.Archive.String()`